BCStudentSoftwareDevTeam/celts
# Reorganized the SLC logic and test suites

### Issue
Resolves #624  

The issue was the SLC logic, controllers, templates, tests, and JavaScript were all pretty poorly organized and 
it was necessarily intuitive to find where things were located. The focus was more on organization rather than code or functionality changes.

### Changes
#### Note: THIS WAS NOT A FUNCTIONALITY CHANGE OVER EVEN MUCH OF A REFACTOR. IF SOMETHING IS BROKEN OR LOOKS WEIRD IN THE FUNCTIONS WE MOVED AROUND IT WAS NOT US.
- Removed the `manageSLFaculty.py` file and moved the functions into `serviceLearningCourses.py`
- Removed the `courseManagement.py` file and moved the functions into `serviceLearningCourses.py`
- Organized the functions in `serviceLearningCourses.py` and added typing to certain functions.
- Modified the imports in routes to reflect our changes in logic.
- Moved the `reviewProposal` logic function from `main/routes.py` to `serviceLearning/routes.py`
- Consolidated `test_manageSLFaculty.py` and `test_courseManagement.py` into `test_serviceLearningCourses.py`.
- Made random fixes, removals, and additions from many different Javascript files.

### To Test
- `git checkout 624_service_learning_refactor` and `git pull origin 624_service_learning_refactor`
- `source setup.sh` and `./database/reset_database.sh test`
- Look over the file structure, make sure the SLC related files are in the places they should be.
- `flask run`
- Open up CELTS and so a quick run through of some key SLC pages, no functionality was directly modified, just want to make sure that everything is imported into routes correctly.
